website/server/models/group.js

```diff
@@ -648,6 +648,27 @@ schema.methods.sendChat = function sendChat (options = {}) {
   return newChatMessage;
 };
 
+schema.methods.handleQuestInvitation = async function handleQuestInvitation (user, accept) {
+  // Handle quest invitation atomically (update only current member when still undecided)
+  // to prevent multiple concurrent requests overriding updates
+  // see https://github.com/HabitRPG/habitica/issues/11398
+  const Group = this.constructor;
+  const result = await Group.update(
+    {
+      _id: this._id,
+      [`quest.members.${user._id}`]: { $exists: true, $eq: null },
+    },
+    { $set: { [`quest.members.${user._id}`]: !!accept } },
+  ).exec();
+
+  if (result.nModified) {
+    // update also current instance so future operations will work correctly
+    this.quest.members[user._id] = !!accept;
+  }
+
+  return !!result.nModified;
```

Comment:
can we use `Boolean(result.nModified)` instead of the two `!!`? it's a bit easier to read, for the two occurances above (`!!accept`), we're converting to a boolean but the function already requires a boolean and being used internally I don't think a forced conversion is necessary. 

What I'd do maybe is throw an error if the user or accept param isn't passed